docs/layering.md

```diff
@@ -0,0 +1,367 @@
+
+```bash
+# Setup the correct SELinux label to allow access to the config
+$ chcon --verbose --type svirt_home_t autologin.ign
+
+# Start a Fedora CoreOS virtual machine
+$ virt-install --name=fcos --vcpus=2 --ram=2048 --os-variant=fedora-coreos-stable \
+    --import --network=bridge=virbr0 --graphics=none \
+    --qemu-commandline="-fw_cfg name=opt/com.coreos/config,file=${PWD}/autologin.ign" \
+    --disk=size=20,backing_store=${PWD}/fedora-coreos.qcow2
+```
+
+The `virt-install` command will start an instance named `fcos` from the `fedora-coreos.qcow2` image using the `autologin.ign` Ignition config. It will auto-attach the serial console of the machine so you will be able to see the image bootup messages.
+
+Once the machine is booted up you should see a few prompts and then you should be automatically logged in and presented with a bash shell:
+
+```
+[  OK  ] Started rpm-ostree System Management Daemon.
+
+Fedora CoreOS 32.20200715.3.0
+Kernel 5.7.8-200.fc32.x86_64 on an x86_64 (ttyS0)
+
+SSH host key: SHA256:XlbayjbgDKNoAAHQxsEL5Q7BdwLxxWSw4NXN9SALLmo (ED25519)
+SSH host key: SHA256:3sx5jseteO4BvdOMWIi0J4koQL015mLonnD0UPTtnZk (ECDSA)
+SSH host key: SHA256:K0fn5/TMJOoMs7Fu7RRkE7IBEf2t8OYCfVaVc+GJWGs (RSA)
+ens2: 192.168.122.127 fe80::5054:ff:feb9:3d97
+Ignition: user provided config was applied
+No ssh authorized keys provided by Ignition or Afterburn
+tutorial login: core (automatic login)
+
+[core@tutorial ~]$
+```
+
+We can verify that our configuration has been applied: we were automatically logged in to the terminal, and the hostname on the prompt is `tutorial`.
+
+As we can see our `/etc/cachefilesd.conf` file:
+
+```
+[core@tutorial ~]$ cat /etc/cachefilesd.conf
+dir /var/cache/fscache
+tag mycache
+brun  8%
+bcull 6%
+bstop 2%
+frun  9%
+fcull 5%
+fstop 4%
+secctx system_u:system_r:cachefiles_kernel_t:s0
+```
+
+Let's also make sure there is no cachefiled package installed.
+```
+[core@tutorial ~]$ rpm -qi cachefilesd
+package cachefilesd is not installed
+```
+
+### Create a derived image with additional content
+
+We are assuming there is a need for debugging on your environment and you need strace and a custom binary. Finally we are going to add keylime and rebase your running FCOS deployment to include them.
+
+This would be helpful when IT needs to provide the images or there is a need to add the same system image to multiple hosts. 
+Instead of making the changes on each host which will have different versions of software or configs you can create an image which then you can rebase your systems on. We are using the [fcos-derivation-example](https://github.com/coreos/fcos-derivation-example) to start this process.
+
+Let's clone our example code and enter the directory.
+
+```bash
+[core@tutorial ~]$ git clone git@github.com:coreos/fcos-derivation-example.git && cd fcos-derivation-example
+```
+
+Examine the Dockerfile and notice the custom binary and where we are installing strace.
```

Comment:
s/Dockerfile/Containerfile maybe?

Reply:
I like that, we would need to update the derivation example too just rename the Dockerfile to Containerfile. @cgwalters WDYT? Any reason to stick to Dockerfile ?

Reply:
I don't have a strong opinion on this myself honestly.  We aren't very consistent about this even in our sub-group; e.g. we use [Dockerfile for coreos-assembler](https://github.com/coreos/coreos-assembler/blob/main/Dockerfile).

I think though honestly I'd lean towards keeping `Dockerfile` in this specific case because it helps send a message that you can build this thing in all well known container build systems.

Reply:
I did not know that a Containerfile was a thing until @travier mentioned it, I like the idea of using the more "not branded" filename but also people could have no idea we are talking about compatible things. Maybe we leave it like this until this has more mind share? You  ok with that @travier ?

Reply:
Sure, this is just a suggestion. I think we should move as much as possible to Containerfile but this is not urgent.